aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/HttpResponseChecksumDecorator.kt

```diff
@@ -65,19 +77,50 @@ class HttpResponseChecksumDecorator : ClientCodegenDecorator {
         baseCustomizations.letIf(applies(operation)) {
             it + HttpResponseChecksumCustomization(codegenContext, operation)
         }
+
+    override fun configCustomizations(
+        codegenContext: ClientCodegenContext,
+        baseCustomizations: List<ConfigCustomization>,
+    ): List<ConfigCustomization> = baseCustomizations + HttpResponseChecksumConfigCustomization(codegenContext)
+
+    /**
+     * Copy the `response_checksum_validation` value from the `SdkConfig` to the client config
+     */
+    override fun extraSections(codegenContext: ClientCodegenContext): List<AdHocCustomization> =
+        if (!serviceHasHttpChecksumOperation(codegenContext)) {
+            listOf()
+        } else {
+            listOf(
+                adhocCustomization<SdkConfigSection.CopySdkConfigToClientConfig> { section ->
+                    rust(
+                        """
+                        ${section.serviceConfigBuilder}.set_response_checksum_validation(${section.sdkConfig}.response_checksum_validation());
+                        """,
+                    )
+                },
+            )
+        }
```

Comment:
obviously it's too late to update this PR now, but in the future we should avoid `if not`s and instead define this and things like it like this:

```
        if (serviceHasHttpChecksumOperation(codegenContext)) {
            listOf(
                adhocCustomization<SdkConfigSection.CopySdkConfigToClientConfig> { section ->
                    rust("""
                        ${section.serviceConfigBuilder}
                            .set_response_checksum_validation(
                                ${section.sdkConfig}.response_checksum_validation()
                            );
                    """)
                },
            )
        } else {
            listOf()
        }
```

Reply:
Updated in https://github.com/smithy-lang/smithy-rs/commit/b80f97148c9d943509d1a8936ce5637da5ab525e